src/modules/Game.class.js

```diff
@@ -6,63 +6,261 @@
+
+    const updatedState = rotateState.map((row) => this.move(row));
+
+    const unRotateSteate = this.rotateMatrixCounteClockwise(updatedState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateSteate)) {
+      this.updateGameState(unRotateSteate);
+      this.addCells();
+    }
+  }
+
+  move(row) {
+    const newRow = [];
+    let i = 0;
+
+    while (i < row.length) {
+      const current = row[i];
+
+      if (current) {
+        let pushed = false;
+
+        for (let j = i + 1; j < row.length; j++) {
+          const next = row[j];
+
+          if (next === current) {
+            newRow.push(current * 2);
+            this.updateGameScore(current * 2);
+            pushed = true;
+            i = j + 1;
+            break;
+          } else if (next) {
+            newRow.push(current);
+            pushed = true;
+            i = j;
+            break;
+          }
+        }
+
+        if (!pushed) {
+          newRow.push(current);
+          i++;
+        }
+      } else {
+        i++;
+      }
+    }
+
+    while (newRow.length < row.length) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  rotateMatrixClockwise(matrix) {
+    const n = matrix.length;
+    const rotatedMatrix = [];
+
+    for (let i = 0; i < n; i++) {
+      rotatedMatrix.push([]);
+
+      for (let j = 0; j < n; j++) {
+        rotatedMatrix[i].unshift(matrix[j][i]);
+      }
+    }
+
+    return rotatedMatrix;
+  }
+
+  rotateMatrixCounteClockwise(matrix) {
+    const n = matrix.length;
+    const rotatedMatrix = [];
+
+    for (let i = 0; i < n; i++) {
+      rotatedMatrix.unshift([]);
+
+      for (let j = 0; j < n; j++) {
+        rotatedMatrix[0].push(matrix[j][i]);
+      }
+    }
+
+    return rotatedMatrix;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.gameStatuses.playing;
+    this.addCells(2);
+  }
+
+  restart() {
+    this.status = Game.gameStatuses.idle;
+    this.resetState();
+  }
+
+  getEmptyCells() {
+    return this.state
+      .flatMap((row, rowIndex) => {
+        return row.map((cell, colIndex) => {
+          return cell === 0 ? [rowIndex, colIndex] : null;
+        });
+      })
+      .filter((cell) => cell !== null);
+  }
+
+  createNewTile() {
+    const emptyCells = this.getEmptyCells();
+
+    if (!emptyCells.length) {
+      return;
+    }
+
+    const [row, col] =
+      emptyCells[Math.trunc(Math.random() * emptyCells.length)];
+
+    this.state[row][col] = Math.random() < 0.9 ? 2 : 4;
+  }
+
+  addCells(count = 1) {
+    for (let i = 0; i < count; i++) {
+      this.createNewTile();
+    }
+
+    const state = this.getState();
+
+    if (this.isVictory(state)) {
+      this.status = Game.gameStatuses.win;
+    } else if (this.isGameOver(state)) {
+      this.status = Game.gameStatuses.lose;
+    }
+  }
+
+  resetState() {
```

Comment:
The 'resetState' function resets the game state to the initial configuration. Ensure that this function is called at the appropriate times, such as when the game is restarted.